### PR TITLE
feat: [BREAKING_CHANGE] env as prod when EGG_SERVER_ENV undefined & NODE_ENV prod

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

https://github.com/eggjs/egg-core/pull/23#issuecomment-254717826

### CODE REVIEW

**Comment by popomore** [comment]
用例挂了

File: lib/loader/egg_loader.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ class EggLoader {
         serverEnv = 'unittest';
       } else if (process.env.NODE_ENV === 'production') {
         console.warn('NODE_ENV is production but EGG_SERVER_ENV is empty, forgot `EGG_SERVER_ENV=prod`? see https://github.com/eggjs/egg-core/blob/master/lib/loader/egg_loader.js#L82 for more detail.');
```
Comment (popomore) [inline lib/loader/egg_loader.js]: 那这个 warn 就没用了？

**Comment by atian25** [comment]
@popomore @fengmk2 done

File: test/loader/get_server_env.test.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ describe('test/loader/get_server_env.test.js', function() {
   it('should use default when NODE_ENV = production', function() {
```
Comment (popomore) [inline test/loader/get_server_env.test.js]: should use prod

Comment (atian25) [inline test/loader/get_server_env.test.js]: done

**Comment by fengmk2** [comment]
+1

**Pull request merged**